csrc/fips_status.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+#include <cstdio>
+#include <functional>
+#include <jni.h>
+
+#include "string_vector.h"
+
+static AmazonCorrettoCryptoProvider::ConcurrentStringVector fipsStatusErrors;
+
+// To have this symbol exported, one needs to modify the final-link.version and the CMakeLists.txt
+extern "C" void AWS_LC_fips_failure_callback(char const* message);
+
+#if defined(FIPS_SELF_TEST_SKIP_ABORT)
+void AWS_LC_fips_failure_callback(char const* message)
+{
+    fprintf(stderr, "AWS_LC_fips_failure_callback invoked with message: '%s'\n", message);
+    fipsStatusErrors.push_back(message);
+}
+#else
+void AWS_LC_fips_failure_callback(char const* message) { }
```

Comment:
Hm, should we call `abort()` here as defense-in-depth? It looks like [AWS-LC will call the symbol](https://github.com/aws/aws-lc/blob/2c1b35b8feeebb4038688d5c04dc0a48421ce06b/crypto/fipsmodule/bcm.c#L416) as long as it's non-null. So, does the empty function "resolve" to `NULL`? If not, we should call `abort()` here.

```suggestion
void AWS_LC_fips_failure_callback(char const* message) {  abort(); }
```

Reply:
I went ahead and did this, leaving thread open for discussion.